File: LeetCode/Medium/1922-count-good-numbers/1922-count-good-numbers.cpp
class Solution {
public:
     int  MOD=1e9+7;
    
    long long p(long long a, long long b){
        if(b==0){
            return 1;
        }

        long long half=p(a,b/2);
        long long ans=(half*half)%MOD;

        if(b%2==1){
            ans=(ans*a)%MOD;
        }
        return ans;
    }
    int countGoodNumbers(long long n) {
        return p(5,(n+1)/2)* p(4,n/2)%MOD;

        
    }
};

File: LeetCode/Medium/1922-count-good-numbers/1922-count-good-numbers.java
class Solution {
    long MOD=1000000007;
    public int countGoodNumbers(long n) {
        long even=(n+1)/2;
        long odd=n/2;
        long res=(findPow(5, even)*findPow(4, odd))%MOD;
        return (int)res;
    }

    public long findPow(long a, long b) {
        if(b==0) {
            return 1;
        }
        long half=findPow(a, b/2);
        long res=(half*half)%MOD;
        if(b%2==1) {
            res=(res*a)%MOD;
        }
        return res;
    }
}